src/nccl_ofi_api.c

```diff
@@ -744,6 +758,25 @@ ncclResult_t nccl_net_ofi_irecv_v4(void* recvComm, void* data, int size,
 	return nccl_net_ofi_irecv(recvComm, 1, &data, &size, &tag, &mhandle, request);
 }
 
+ncclResult_t nccl_net_ofi_irecv_v9(void* recvComm, int n, void** data,
+				size_t* sizes, int* tags, void** mhandles, void** request)
+{
+    int sizesInt[NCCL_PROXY_MAX_SUBS];
+    ncclResult_t validation_result = ncclSuccess;
+    //reset to NULL if optional receive completion is set
+    if (*request == (void *) NCCL_NET_OPTIONAL_RECV_COMPLETION) {
+        *request = NULL;
+    }
+    for (int i = 0; i < n; i++) {
+        validation_result = is_size_valid(sizes[i]);
```

Comment:
Why not have the function signature be `msg_length_verify_max_size(size_t sizes, size_t len)` and call it here as `if (ncclSuccess != msg_length_verify_max_size(sizes, n));` and in send as `if (ncclSuccess != msg_length_verify_max_size(&size, 1)`?